pymeasure/instruments/hp/hp8657b.py

```diff
@@ -0,0 +1,200 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from enum import IntEnum
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class HP8657B(Instrument):
+    """ Represents the Hewlett Packard 8657B signal generator
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Hewlett-Packard HP8657B",
+            includeSCPI=False,
+            send_end=True,
+            **kwargs,
+        )
+
+    class Modulation(IntEnum):
+        """
+        IntEnum for the different modulation sources
+        """
+        EXTERNAL = 1
+        INT_400HZ = 2
+        INT_1000HZ = 3
+        OFF = 4
+        DC_FM = 5
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+        as the 8657B does not support any readout of values, this will return 0 and log a warning
+
+        """
+        log.warning("HP8657B Does not support error status readout")
+
+    def clear(self):
+        """
+        Reset the instrument to power-on default settings
+
+        """
+        self.adapter.connection.clear()
+
+    id = "HP 8657B Signal generator"  #: Manual ID entry, as the instrument does not talk with us
```

Comment:
@LongnoseRob, nice idea to have an `id` anyway for uniformity with other scpi instruments.
I would consider to add the string in IEEE 488.2 format.
I report here an excerpt from a documentation for an Anritsu instrument.

`identification query. query only. this query returns an instrument identification string in ieee-488.2 specified <arbitrary ascii> format consisting of four fields separated by commas. the fields are: <manufacturer>, <model>, <serial #>,<firmware revision level>`

So, in your case, it should be something like:
`"HP,8657B,N/A,N/A"`

What do you think about this?
If there is a general agreement, we could put this as suggestion in PR #712.

Reply:
> So, in your case, it should be something like:
"HP,8657B,N/A,N/A"
 What do you think about this?

Thanks for the feedback and the improvement idea.
One question the _implementation_, would t 
```
id = "HP,8657B,N/A,N/A"
```
be sufficient or should it be 

```
def id(self):
    """"FAKE *IDN? return"""
    return "HP,8657B,N/A,N/A"
```

Reply:
Well, I think that property exists to allow access to a value that requires some calculation (i.e. code execution) like an attribute. For example:
```
@property
    def id(self):
        """ Requests and returns the identification of the instrument. """
        if self.SCPI:
            return self.ask("*IDN?").strip()
        else:
            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
```
In our case, we don't need to run any code, so we can go back to plain attribute assignment.